migration_scripts/bulk_load/src/footprint.clj

```diff
@@ -309,3 +622,90 @@
              :tx-kinds
              :tx-recipients
              :tx-senders])))
+
+;; KV Store ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn offloaded-to-kv-store
+  "Simulate offloading blob data to the key-value store.
+
+  - Columns that have been marked to be moved into a key-value
+    store (according to `kv-columns`) have their sizes removed from
+    their corresponding table and gathered into a special `:kv-store`
+    entry in the sizes mapping.
+
+  - It's assumed that if a table has columns to be moved to a KV
+    store, all its TOAST-ed values belong to those columns, and will
+    have their size re-attributed to the `:kv-store`.
+
+  - It's also assumed that these columns will not impact indices and
+    therefore won't change index or primary key size, unless the entire
+    table can be moved to key-value store."
+  [tables]
+  (let [kv-size (atom 0)
+
+        ;; Indicate whether the column should be moved to the KV
+        ;; store (return `:mv`), removed from RDBMS (`:rm`),
+        ;; copied (`:cp`), or is not affected by offloading (`nil`).
+        kv-column
+        (fn [table col]
+          (let [fq #(keyword (name table) (str (name col) %))]
+            (cond
+              (kv-columns (fq ""))  :mv
+              (kv-columns (fq "*")) :rm
+              (kv-columns (fq "+")) :cp)))
+
+        ;; Accepts a per-table statistic, and tries to extract
+        ;; key-value data from it. Returns the updated statistic, and
+        ;; updates `kv-size` with the size that should be attributed
+        ;; to the key-value store.
+        extract-kv
+        (fn [table {:keys [cols tuples toast self] :as stat}]
+          (cond
+            ;; If we don't have column or tuple information or it
+            ;; indicates that there are no key-value columns in this
+            ;; table, skip it.
+            (or (not cols) (not tuples)) stat
+            (not-any? #(kv-column table %) (keys cols)) stat
```

Comment:
from earlier, we don't generate per-col info for events, transactions, and objects tables, so it seems like we're missing out on a lot of simulation data w.r.t offloading to kv store?

Reply:
(see my note on that PR -- we do get column stats for those tables, but they are scoped at the per-partition level).